blog/_posts/2020-04-20-first-release.adoc

```diff
@@ -0,0 +1,200 @@
+
+video::/assets/blog/first-release/extend-selection.webm[options="autoplay,loop",width=100%]
+
+Fixit for missing module::
+
+Another disproportionally nice feature -- to create a new file, type `mod file_name;` and use kbd:[ctrl+.] to add the file itself.
+
+video::/assets/blog/first-release/new-mod.webm[options="autoplay,loop",width=100%]
+
+Assists::
+
+More generally, there are a lot of cases where the light bulb can write some boring code for you.
+Some of my favorites are impl generation:
+
+video::/assets/blog/first-release/impl-iter.webm[options="autoplay,loop",width=100%]
+
+And filling match arms:
+
+video::/assets/blog/first-release/fill-match-arms.webm[options="autoplay,loop",width=100%]
+
+== Drawbacks
+
+Of course, rust-analyzer being a young tool, it comes with a lot of limitations.
+The most significant one is that we are not at the moment using `rustc` directly, so our capabilities for detecting errors are limited.
+
+In particular, to show inline errors we are doing what Emacs has been doing for ages -- running `cargo check` after the file is saved.
+if the auto-save is enabled in the editor, the result is actually quite nice for small projects.
+
+For bigger projects though, I feel like `cargo check` in background gets in a way.
+So for `rust-analyzer` I have `rust-analyzer.checkOnSave.enabled = false;` in settings.
+Instead, I use the **Run** functionality to run `check` / `test` and keyboard shortcuts to navigate between errors.
+
+Another big issue is that at the moment we, for simplicity, don't persist caches to disk.
+That means that every time you open a new project with rust-analyzer, it needs to analyzer, from source:
+
+* all sysroot crates (std, core, alloc, etc)
+* all crates.io dependencies
+* all crates in your workspace
+
+This takes time, tens of seconds for medium sized projects.
+
+Similarly, because we never save anything to disk, we need to keep analysis results for all crates in memory.
+At the moment, rust-analyzer process requires gigabytes of ram.
+
+Finally, because analysis is not complete, features are not working correctly every time.
+Sometimes there are missing completions, sometimes goto definition is wrong, we may even show false-positive errors on occasion.
+
+This is an alpha release.
+We have a long road ahead of us towards solid and reliable IDE support.
+Luckily (and this is the instance where a life of IDE writer is simpler than that of compiler writer) and IDE doesn't have to be 100% correct to be useful.
+
+== How can I help?
+
+If you find rust-analyzer useful and you use it professionally, consider asking your company to sponsor rust-analyzer.
+We accept sponsorship at https://opencollective.com/rust-analyzer/.
+You can also write directly to office@ferrous-systems.com.
+We, of course, are also open to individual backers, but we'd love to focus on corporate sponsorship.
```

Comment:
maybe also mention that people can contribute code? ;)

Reply:
Yeah, I've run out of the fuel by the end, I hope to revisit this section before Monday. I am hesitant a bit about direct call for participation, as I don't feel good about our current review capacity...

Reply:
"but we'd love to focus on corporate sponsorship." — I would name the reason openly: processing costs make corporate sponsorships more efficient (or smth along these lines)